addons/point_of_sale/models/pos_config.py

```diff
@@ -8,33 +8,66 @@
-
-    @api.multi
-    def name_get(self):
-        result = []
-        for cashbox_line in self:
-            result.append((cashbox_line.id, "%s * %s"%(cashbox_line.coin_value, cashbox_line.number)))
-        return result
-
 class AccountBankStmtCashWizard(models.Model):
     _inherit = 'account.bank.statement.cashbox'
 
+    @api.depends('pos_config_ids')
+    def _compute_currency(self):
+        super(AccountBankStmtCashWizard, self)._compute_currency()
+        for cashbox in self:
+            if cashbox.pos_config_ids:
+                cashbox.currency_id = cashbox.pos_config_ids[0].currency_id.id
+
+    currency_id = fields.Many2one('res.currency', compute='_compute_currency')
+    pos_config_ids = fields.One2many('pos.config', 'default_cashbox_id')
+    is_a_template = fields.Boolean(default=False)
+
     @api.model
     def default_get(self, fields):
         vals = super(AccountBankStmtCashWizard, self).default_get(fields)
+        if "is_a_template" in fields and self.env.context.get('default_is_a_template'):
+            vals['is_a_template'] = True
         config_id = self.env.context.get('default_pos_id')
         if config_id:
-            lines = self.env['account.cashbox.line'].search([('default_pos_id', '=', config_id)])
+            config = self.env['pos.config'].browse(config_id)
+            previous_session_list = self.env['pos.session'].search([('config_id', '=', config_id)])
+            if len(previous_session_list) > 1 and previous_session_list[1].cash_register_id.cashbox_end_id.cashbox_lines_ids:
+                lines = previous_session_list[1].cash_register_id.cashbox_end_id.cashbox_lines_ids
+            else:
+                lines = config.default_cashbox_id.cashbox_lines_ids
             if self.env.context.get('balance', False) == 'start':
                 vals['cashbox_lines_ids'] = [[0, 0, {'coin_value': line.coin_value, 'number': line.number, 'subtotal': line.subtotal}] for line in lines]
-            else:
+            else:	
                 vals['cashbox_lines_ids'] = [[0, 0, {'coin_value': line.coin_value, 'number': 0, 'subtotal': 0.0}] for line in lines]
         return vals
 
+    def confirm_cash_box(self):
+        current_session = self.env['pos.session'].browse(self.env.context['pos_session_id'])
+        self._set_bank_statement_amount()
+        if current_session.state == 'new_session':
+            current_session.write({'state': 'opening_control'})
+
+    @api.multi
+    def set_default_cashbox(self):
+        current_session = self.env['pos.session'].browse(self.env.context['pos_session_id'])
+        lines = current_session.config_id.default_cashbox_id.cashbox_lines_ids
+        context = dict(self._context)
+        self['cashbox_lines_ids'] = False
+        self['cashbox_lines_ids'] = [[0, 0, {'coin_value': line.coin_value, 'number': line.number, 'subtotal': line.subtotal}] for line in lines]
```

Comment:
it should be list of tuple, not list of list [(0, 0, {'coin_value': line.coin_value, 'number': line.number, 'subtotal': line.subtotal})]

You should not self['cashbox_lines_ids'] = False, because you want to do that you can just start your list of command with a command (5, 0, 0) and you don't delete the old lines.

also avoid writing this: self['cashbox_lines_ids'] -> self.cashbox_lines_ids

So first delete existing, then create the new ones:
self.cashbox_lines_ids.unlink()
self.cashbox_lines_ids = [[0, 0, {'coin_value': line.coin_value, 'number': line.number, 'subtotal': line.subtotal}] for line in lines]